packages/widget/src/components/Switch.tsx

```diff
@@ -31,18 +31,14 @@ type SwitchProps = {
   onChange?: (val: boolean) => void;
 };
 
-export const Switch = ({ checked = false, onChange }: SwitchProps) => {
-  const [isChecked, setIsChecked] = useState(checked);
-
+export const Switch = ({ checked, onChange }: SwitchProps) => {
   const toggle = () => {
-    const newVal = !isChecked;
-    setIsChecked(newVal);
-    onChange?.(newVal);
+    onChange?.(!checked);
   };
 
   return (
-    <SwitchWrapper checked={isChecked} onClick={toggle}>
-      <SwitchThumb checked={isChecked} />
+    <SwitchWrapper checked={!!checked} onClick={toggle}>
+      <SwitchThumb checked={!!checked} />
```

Comment:
@codingki i dont think the !!checked is necessary this should be fine?

```
export const Switch = ({ checked = false, onChange }: SwitchProps) => {
  const toggle = () => {
    onChange?.(!checked);
  };

  return (
    <SwitchWrapper checked={checked} onClick={toggle}>
      <SwitchThumb checked={checked} />
    </SwitchWrapper>
  );
};
```

Reply:
yeah that works